tests/unit/acquisition/test_function.py

```diff
@@ -461,6 +471,175 @@ def prepare_acquisition_function(
     npt.assert_allclose(eci(x), ei(x) * pof(x))
 
 
+class _PseudoTrainableQuadratic(QuadraticMeanAndRBFKernel, PseudoTrainableProbModel):
+    pass
```

Comment:
can't resist adding sth here: you shouldn't need `PseudoTrainableProbModel` when testing acquisition functions, because acquisition functions don't train the model. You can just use `QuadraticMeanAndRBFKernel`

Reply:
When I earlier directly make use of `QuadraticMeanAndRBFKernel`, I got some type checking error (shown [here](https://github.com/secondmind-labs/trieste/runs/2344041151)) as `ModelStack` receive `TrainableProbabilisticModel`,  is there anything that am not doing properlly?

Reply:
you're right, I hadn't thought of that. Perhaps it would be good to have a version of `ModelStack` for non-trainable `ProbabilisticModel`s. Something to think about after this PR

Reply:
aha, I've just remembered this is how I wrote `GaussianProcess`. The mean and kernel lists are one for each output dimension, so you can do sth like
```python
def _mo_test_models(*kernel_amplitudes: float | TensorType | None = None) -> GaussianProcess:
    means = [quadratic, lambda x: tf.reduce_sum(x, axis=-1, keepdims=True), quadratic]
    kernels = [tfp.math.psd_kernels.ExponentiatedQuadratic(k) for k in kernel_amplitudes]
    return GaussianProcess(means, kernels)
```
You don't even need a `ModelStack` for the tests

I'm now just saying things so you know more about the functionality available

Reply:
Ah I see, I was earlier bit trapped with only using `ModelStack` there.
This is already much neater than the current cumbersome test model, will use sth like this indeed, thanks!

Reply:
Removed the `TrainableProbabilisticModel` by using a modified `_mo_test_model` here (Line 468) instead for mo test.